test/options.cpp

```diff
@@ -83,19 +89,36 @@ TEST_CASE("Basic options", "[options]")
   CHECK(result.count("6") == 1);
   CHECK(result.count("p") == 2);
   CHECK(result.count("space") == 2);
-
   auto& arguments = result.arguments();
   REQUIRE(arguments.size() == 7);
   CHECK(arguments[0].key() == "long");
   CHECK(arguments[0].value() == "true");
   CHECK(arguments[0].as<bool>() == true);
-
   CHECK(arguments[1].key() == "short");
   CHECK(arguments[2].key() == "value");
   CHECK(arguments[3].key() == "av");
-
   CHECK_THROWS_AS(result["nothing"].as<std::string>(), cxxopts::option_has_no_value_exception&);
 
+  auto str_result = options.parse(argv.argv_str());
```

Comment:
I don't think duplicating all of the test cases is a good idea. If you really wanted to test both you might be able to use a catch `SECTION` to do it without duplication. But then I would be ok with factoring out the string splitting into its own class and just testing that splitting the strings up gives the correct results.